Base/QTGUI/qSlicerMouseModeToolBar.cxx

```diff
@@ -383,9 +383,44 @@ void qSlicerMouseModeToolBarPrivate::updateCursor()
   int currentInteractionMode = interactionNode->GetCurrentInteractionMode();
   if (currentInteractionMode != vtkMRMLInteractionNode::Place)
     {
-    interactionNode->SetCurrentInteractionMode(vtkMRMLInteractionNode::Place);
+    if (interactionNode->GetCurrentInteractionMode() == vtkMRMLInteractionNode::ViewTransform)
+      {
+      q->changeCursorTo(QCursor());
```

Comment:
How is it possible that we add all this code for setting the cursor here and we do not remove from anywhere else? Where was it originally implemented? Can that code be removed?

Reply:
This change restores the code to update the cursor shape where previously implemented, with a few changes. I removed it while testing since it no longer worked when the actions menu was removed from the mouse mode toolbar. With the markup node displayed in the toolbar, it did not seem as important to provide the visual cue to the placement node type with the cursor, so I did not add it back immediately. I'm still not sure it's necessary for control point placement, but agree that its needed for the WindowLevel action 

In this commit, I've updated the original code to update the cursor shape from the selection node instead of the mouse mode place actions menu.

Reply:
OK, thanks for the explanation.